mlir/lib/IR/AffineExpr.cpp

```diff
@@ -237,6 +239,135 @@ unsigned AffineDimExpr::getPosition() const {
   return static_cast<ImplType *>(expr)->position;
 }
 
+/// Returns true if this expression is made out of only symbols.
+static bool isSymbolic(AffineExpr expr) {
+  if (expr.getKind() == AffineExprKind::SymbolId)
+    return true;
+  return false;
```

Comment:
```return expr.getKinds() == AffineExprKind::SymbolId;``` will just work. But you don't need a function just for this, you already have `isa`. You could just do inline `expr.isa<AffineSymbolExpr>()`.